### ISSUE TITLE
Baseline incorrect with Chinese characters

### ISSUE THREAD

**Issue by archcst** [mgieseki/dvisvgm#253]
### Reproduce the issue
1. `xelatex -no-pdf -output-directory . math.tex` to get the `.xdv` file
2. `xdvipdfmx math.xdv -o math.pdf` to get the `.pdf` file
3. `dvisvgm math.xdv -e -n -b min -o math.svg` to get the `.svg` file

**With comparing the `.pdf` and the `.svg` file, you could find the Chinese characters is lower of English ones:**

<img width="181" alt="SCR-20231102-pyep" src="https://github.com/mgieseki/dvisvgm/assets/41679407/ebcc7c75-63c1-49fa-bc3b-791f257b203c">

### `.tex` file content
```latex
\documentclass{article}  
\usepackage{fontspec}
\setmainfont{华文仿宋}
\begin{document}

$$ \frac{ABC中文}{ABC中文} $$

\end{document}
```

**Comment by mgieseki** [comment]
Could you please provide the font 华文仿宋 or post a link where it can be downloaded? Currently, I can't test your example because of the missing font. 
If possible, also try to convert your file with the latest version 3.1.2 of dvisvgm and check whether the issue occurs there as well.

**Comment by mgieseki** [comment]
The issue might be the same as #235 which has been fixed in dvisvgm 3.1.

**Comment by archcst** [comment]
Thanks for your fast response.
Is there anyway upgrade dvisvgm via Hombrew?

**Comment by mgieseki** [comment]
Unfortunately, I don't know since I'm not familiar with Homebrew. Maybe you could ask on their [discussion page](https://github.com/orgs/Homebrew/discussions) if they provide a more recent version.

**Comment by archcst** [comment]
I’ll try to compile the latest version, you may close this one, thanks again

**Comment by mgieseki** [comment]
Ok, thanks for the update. Please let me know if the latest version fixes the problem for you. Otherwise, I'll have a look what's actually going on.

**Comment by archcst** [comment]
Hi, I'm encountering a compile issue with the 3.1.2 release. Could you please help on this?

```
Undefined symbols for architecture arm64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [hashcheck] Error 1
make[2]: *** [check-am] Error 2
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
```

**Comment by mgieseki** [comment]
Could you please attach the entire build log including the output of `configure` and `make`? The single linker error message doesn't provide enough information to determine the cause of the failure.

**Comment by archcst** [comment]
[log.txt](https://github.com/mgieseki/dvisvgm/files/13249050/log.txt)
Please find log in the attachment, tks

**Comment by mgieseki** [comment]
The dvisvgm binary seems to be built correctly and should be located in the `src` directory. However, the test `hashcheck` can't be built, probably because the corresponding source file `hashcheck.cpp` wasn't created correctly. Maybe it's empty on your system. 
The file should be created during the build process by `genhashcheck.py` which requires Python and the Python package `xxhash` to be installed. 
You can try to build it manually by calling `make hashcheck` inside directory `tests`. Afterwards have a look at the file and check if the `main` function is present there. If not, some Python stuff is probably missing on your computer.

**Comment by mgieseki** [comment]
Yes, that should work. On Linux, `python` is usually a symbolic link to the `python3` executable:
```
> ls -la /usr/bin/python
lrwxrwxrwx. 1 root root 9  3. Okt 02:00 /usr/bin/python -> ./python3
```

**Comment by archcst** [comment]
I found the `dvisvgm` file and mv it to `/usr/local/bin`, now it works fine with 3.12 version, thanks a lot!

![SCR-20231105-kfdt](https://github.com/mgieseki/dvisvgm/assets/41679407/486f1018-9aba-4719-a6ed-79c8f6956310)

but I get this message, 

```shell
❯  dvisvgm --version
dvisvgm 3.1.2

❯  dvisvgm math.xdv -e -n -b min -o math.svg
processing of PostScript specials is disabled (Ghostscript not found)
pre-processing DVI file (format version 7)
processing page 1
  graphic size: 43.44931pt x 568.400711pt (15.270686mm x 199.770002mm)
  output written to math.svg
1 of 1 page converted in 0.190968 seconds
```

It's wired I'm pretty sure I have installed `Ghostscript`

```shell
❯  brew list | grep ghostscript
ghostscript
```

Any hints?

**Comment by mgieseki** [comment]
Great to hear that the latest version fixes the issue for you.

> It's wired I'm pretty sure I have installed `Ghostscript`

dvisvgm requires the Ghostscript library, the Ghostscript executable is not sufficient. You need to make sure that the library is accessible through the search path of the dynamic linker or specify the path with option `--libgs` or environment variable `LIBGS`. Since I don't know much about MacOs, I can't give more specific hints. Here you'll find some more information:
* #143
* https://tex.stackexchange.com/questions/559640/making-dvisvgm-and-ghostscript-from-mactex-talk-to-each-other
* https://tobywf.com/2018/01/beautiful-svg-diagrams-with-tikz-dvisvgm

**Comment by archcst** [comment]
Thanks a lot, all problem solved!

```shell
❯  export LIBGS=/opt/homebrew/Cellar/ghostscript/10.02.0/lib/libgs.10.02.dylib

❯  dvisvgm math.xdv -e -n -b min -o math.svg
pre-processing DVI file (format version 7)
processing page 1
  graphic size: 43.44931pt x 568.400711pt (15.270686mm x 199.770002mm)
  output written to math.svg
1 of 1 page converted in 2.23617 seconds
```
